### DISCUSSION TITLE
Proxy does not terminate SSE connections when the backend closes

### DISCUSSION THREAD

**Discussion by gitpushdashf** [chimurai/http-proxy-middleware#765] [General]
I have a Flask server doing SSE, which I am proxying to. It works fine, until the SSE endpoint dies (which happens on reload). It then does not reconnect on the backend, nor disconnect the client.

If it would boot off the client or reconnect silently on the backend, either would be fine. I'd probably prefer booting off the client in this case, and forcing the client to cause the reconnection.

I've tried some options here and have not had any luck.

Thank you!

**Comment by chimurai** [comment]
Can you share the options you've tried?
If possible, could you share a reproduction repo?

**Reply by gitpushdashf** [reply]
Thanks for getting back to me!

I tried onError, onProxy, proxyTimeout, etc.

https://github.com/gitpushdashf/discussion-765

README has some instructions. Let me know if I can clarify anything.

**Reply by chimurai** [reply]
Think I've found it. Thanks to your repro 👍

Could you try this configuration?

```js
onProxyRes: (proxyRes, req, res) => {
    proxyRes.on('close', () => {
      if (!res.writableEnded) {
        res.destroy();
      }
    });
}
```

**Reply by gitpushdashf** [reply]
Yes! That works perfectly. Thank you so much!

Should I always use this configuration, or is this something that you are planning a fix for? I'm not sure if it's a bug or the default behavior is correct.

**Reply by chimurai** [reply]
Not sure yet whether to include it by default or as a separate "sse" plugin.

Found out this fix conflicts with: https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/response-interceptor.md

Something to investigate what the impact is.

**Comment by nBerg** [comment]
@chimurai I'm running into the same issue. The configuration you shared here works great. 

However I did notice https://github.com/chimurai/http-proxy-middleware/blob/master/src/plugins/default/debug-proxy-errors-plugin.ts#L30 which looks like it might have been fixed in v3. I was looking into upgrading to v3 and I thought I might be able to remove this config but when testing that does not seem to be the case.

I'm just wondering if this issue has been fixed in v3?

**Reply by nBerg** [reply]
Ignore this, I believe I misunderstood the code I linked. If I understand correctly it's handling the opposite error (connection closing on client being properly closed with the API), and that is indeed what the linked tickets described.
